### PR TITLE
Filebeat remove python phase 1

### PR DESCRIPTION
**Pull request by leehinman**
## Proposed commit message

Begin the process of removing python integration tests from filebeat and replacing them with go integration tests.

### infrastructure helpers (new files):

• config.go — FilestreamInputConfig builder with FilestreamOptions (ScanInterval, IgnoreOlder, CloseInactive, Symlinks, ExcludeFiles, NDJSON, Multiline, GlobalProcessors, etc.)
• files.go — WriteFile, AppendToFile, RotateFile, TruncateFile, CreateSymlink
• wait.go — WaitUntil generic polling helper used throughout Phase 2 tests
• helper.go (extended) — RegistryEntry struct, ReadRegistry, RegistryLogPath for full registry state inspection

### new Go integration tests (all use //go:build integration):

• base_test.go — TestBase/BasicFieldsExist (core ECS fields present), TestBase/ShutdownNoInputs (exit code 1 with no inputs)
• input_test.go — ignore_older, exclude_files, close_inactive, symlink skipping, file rotation, files-added-late
• crawler_test.go — multiple appends, follow open file, UTF-8 characters, file disappear/appear
• json_test.go — Docker log parsing, line filtering on NDJSON, keys_under_root, overwrite_keys, add_error_key
• processors_test.go — drop_fields, include_fields, drop_event with path condition

All new tests use the filestream input and output.console — no Elasticsearch dependency.

Removal of converted Python tests from filebeat/tests/system/:

• test_base.py — test_base removed; test_index_management (requires ES) kept
• test_input.py — 8 converted tests removed; 14 unconverted tests kept (rotation edge cases, harvester_limit, identity tracking, recursive glob, etc.)
• test_crawler.py — 4 converted tests removed; test_tail_files, test_encodings, test_file_no_permission kept
• test_json.py — 3 converted tests removed; 11 unconverted tests kept (timestamp parsing, document ID, CSV/URL decode, etc.)
• test_processors.py — 3 converted tests removed; 10 unconverted tests kept (dissect, truncate, CSV, URL decode, JavaScript processor)

## WHY:

Maintaining two languages is a burden and consolidating onto one language reduces our maintenance overhead.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works. Where relevant, I have used the [`stresstest.sh`](https://github.com/elastic/beats/blob/main/script/stresstest.sh) script to run them under stress conditions and race detector to verify their stability.
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent-changelog-tool/blob/main/docs/usage.md).

## Disruptive User Impact

None. Test code

## How to test this PR locally

```
cd filebeat && mage goIntegTest
```

### CODE REVIEW

File: filebeat/testing/integration/base_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	libbeatintegration "github.com/elastic/beats/v7/libbeat/testing/integration"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// TestBase covers the fundamental tests from test_base.py.
```
Comment (orestisfl) [inline filebeat/testing/integration/base_test.go]: Do we want to keep python names around in comments?

File: filebeat/testing/integration/crawler_test.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	require.NoError(t, err)
+	require.NoError(t, f.Sync())
+	f.Close()
+
+	// The beat is killed once "hello world 1" and "hello world 2" are seen.
+	test.Wait()
+}
+
+// TestCrawlerUTF8 verifies that UTF-8 characters are read correctly.
+// Corresponds to test_crawler.py::test_utf8.
```
Comment (orestisfl) [inline filebeat/testing/integration/crawler_test.go]:
```suggestion
```

File: filebeat/testing/integration/crawler_test.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	WaitUntil(t, func() bool { return firstCount.Load() >= 1 }, 15*time.Second, 100*time.Millisecond)
+
+	// Delete the original file; Filebeat should notice the removal and close it.
+	require.NoError(t, os.Remove(logFile))
+
+	// Wait for filebeat to log that it closed the removed file before creating a new one
+	// at the same path.  A 500ms wait is a pragmatic minimum; on slow CI hosts
+	// this may occasionally be too short, but close_removed fires on the next
+	// scan tick (100ms), so 500ms provides 5 scan opportunities.
+	time.Sleep(500 * time.Millisecond)
```
Comment (orestisfl) [inline filebeat/testing/integration/crawler_test.go]: Here and  everywhere with `time.Sleep`: can we wait for filebeat logs as we do in other integration tests? If it's not easy, can we track it as a follow-up?

File: filebeat/testing/integration/input_test.go [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+	libbeatintegration "github.com/elastic/beats/v7/libbeat/testing/integration"
+)
+
+var inputReportOptions = libbeatintegration.ReportOptions{
+	PrintLinesOnFail:  100,
+	PrintConfigOnFail: true,
+}
+
+// TestInputIgnoreOlderFiles verifies that files not modified within ignore_older
+// are skipped.  Corresponds to test_input.py::test_ignore_older_files.
```
Comment (orestisfl) [inline filebeat/testing/integration/input_test.go]:
```suggestion
// are skipped.
```

**Review by orestisfl — APPROVED** [review]
Approving to unblock, some nitpicks.

File: filebeat/testing/integration/crawler_test.go [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build integration
+
+// Tests for file-discovery and crawling behaviour, ported from test_crawler.py.
```
Comment (khushijain21) [inline filebeat/testing/integration/crawler_test.go]:
```suggestion
// Tests for file-discovery and crawling behaviour
```

File: filebeat/testing/integration/crawler_test.go [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+)
+
+var crawlerReportOptions = libbeatintegration.ReportOptions{
+	PrintLinesOnFail:  100,
+	PrintConfigOnFail: true,
+}
+
+// TestCrawlerMultipleAppends verifies that Filebeat keeps picking up new lines
+// as they are appended to a file in several rounds.
+// Corresponds to test_crawler.py::test_multiple_appends.
```
Comment (khushijain21) [inline filebeat/testing/integration/crawler_test.go]:
```suggestion

```

File: filebeat/testing/integration/files.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+package integration
+
+import (
+	"os"
+	"testing"
+)
+
+// WriteFile writes content to path, creating the file if needed and truncating any existing content.
+func WriteFile(t *testing.T, path, content string) {
```
Comment (khushijain21) [inline filebeat/testing/integration/files.go]: I think we can use existing methods instead of creating new helpers

[See](https://github.com/khushijain21/beats/blob/cc2b445e718da356248e7a2ffb2a223fbefdbe3b/filebeat/testing/integration/log_generator.go#L96)

File: filebeat/testing/integration/files.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+	t.Helper()
+	if err := os.WriteFile(path, []byte(content), 0644); err != nil {
+		t.Fatalf("WriteFile(%q): %v", path, err)
+	}
+}
+
+// AppendToFile appends content to path, creating the file if it does not exist.
+func AppendToFile(t *testing.T, path, content string) {
```
Comment (khushijain21) [inline filebeat/testing/integration/files.go]: [See](https://github.com/khushijain21/beats/blob/cc2b445e718da356248e7a2ffb2a223fbefdbe3b/filebeat/testing/integration/log_generator.go#L159)

File: filebeat/testing/integration/files.go [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+		t.Fatalf("AppendToFile open(%q): %v", path, err)
+	}
+	defer f.Close()
+	if _, err := f.WriteString(content); err != nil {
+		t.Fatalf("AppendToFile write(%q): %v", path, err)
+	}
+}
+
+// RotateFile renames src to dst, simulating a log rotation.
+func RotateFile(t *testing.T, src, dst string) {
```
Comment (khushijain21) [inline filebeat/testing/integration/files.go]: [nit] We can move these helper methods under `helper.go` file instead

File: filebeat/testing/integration/base_test.go (hunk 1 above)
Comment (leehinman) [reply filebeat/testing/integration/base_test.go]: removed

File: filebeat/testing/integration/crawler_test.go (hunk 3 above)
Comment (leehinman) [reply filebeat/testing/integration/crawler_test.go]: replaced

File: filebeat/testing/integration/files.go (hunk 7 above)
Comment (leehinman) [reply filebeat/testing/integration/files.go]: done

File: filebeat/testing/integration/files.go (hunk 8 above)
Comment (leehinman) [reply filebeat/testing/integration/files.go]: done

File: filebeat/testing/integration/files.go (hunk 9 above)
Comment (leehinman) [reply filebeat/testing/integration/files.go]: removed files.go
